src/dsp_tools/commands/xmlupload/models/serialise/serialise_value.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+from typing import Optional
+from typing import Protocol
+
+
+@dataclass
+class SerialiseProperty:
+    property_name: str
+    values: list[SerialiseValue]
+
+    def serialise(self) -> dict[str, Any]:
+        """Serialise the property and all its values."""
+        return {self.property_name: [value.serialise() for value in self.values]}
+
+
+@dataclass
```

Comment:
is it actually meaningful to mark a protocol as dataclass? to me this sounds contradictory, but I may be wrong

Reply:
mypy doesn't accept the duck-typing if I don't declare this as a dataclass too...

Reply:
fair enough

Reply:
Now with explicit subtyping à la Christian it works without adding the decorator